src/probnum/diffeq/odefiltsmooth/odefiltsmooth.py

```diff
@@ -223,42 +224,45 @@ def probsolve_ivp(
      [0.9794734 ]
      [0.98614926]]
     """
-    gfilt, firststep, stprl = _create_solver_inputs(
-        ivp, method, which_prior, tol, step, firststep, precond_step, **kwargs
+    _check_step_tol(step, atol, rtol)
+    gfilt, stprl = _create_solver_inputs(
+        ivp, method, which_prior, step, firststep, precond_step, **kwargs
     )
     with_smoothing = method[-2] == "s" or method[-1] == "s"
     solver = GaussianIVPFilter(ivp, gfilt, with_smoothing=with_smoothing)
-    solution = solver.solve(firststep=firststep, steprule=stprl, **kwargs)
+    solution = solver.solve(steprule=stprl, atol=atol, rtol=rtol, **kwargs)
```

Comment:
This is how it was before, though I find it a bit awkward to save the steprule in the solver. The decision of which solver to use is kind of orthogonal to the decision which steprule to use, or what do you think? I also think it is a bit unnecessary to let the ODEsolver carry a steprule around even though it is only needed in solve. What do you think?

Reply:
> The decision of which solver to use is kind of orthogonal to the decision which steprule to use, or what do you think?

The vocabulary is not so precise here I think. If "solver" means Euler vs RK4 vs ODE filter then I totally agree! But I think one can also think about the "solver" as the full procedure going from problem to a solution estimate, which includes the step selection algorithm, so it's not so clear for me.

> I also think it is a bit unnecessary to let the ODEsolver carry a steprule around even though it is only needed in solve. What do you think?

This is a really good argument. If the steprule is only used in a single function of the solver, then it's hard to justify saving it in the solver itself. So in summary I agree with the current implementation :)

Reply:
> The vocabulary is not so precise here I think. If "solver" means Euler vs RK4 vs ODE filter then I totally agree! But I think one can also think about the "solver" as the full procedure going from problem to a solution estimate, which includes the step selection algorithm, so it's not so clear for me.

I meant solver as in `ODESolver`. You're right that this can mean the entire process, though here, I wasn't thinking in such general terms :)